internal/shardchain/local_cache/local_cache_database.go

```diff
@@ -0,0 +1,112 @@
+package local_cache
+
+import (
+	"bytes"
+	"log"
+	"time"
+
+	"github.com/allegro/bigcache"
+	"github.com/ethereum/go-ethereum/ethdb"
+)
+
+type cacheWrapper struct {
+	*bigcache.BigCache
+}
+
+func (c *cacheWrapper) Put(key []byte, value []byte) error {
+	return c.BigCache.Set(String(key), value)
+}
+
+func (c *cacheWrapper) Delete(key []byte) error {
+	return c.BigCache.Delete(String(key))
+}
+
+type LocalCacheDatabase struct {
+	ethdb.KeyValueStore
+
+	enableReadCache bool
+
+	deleteMap map[string]bool
+	readCache *cacheWrapper
+}
+
+func NewLocalCacheDatabase(remoteDB ethdb.KeyValueStore) *LocalCacheDatabase {
+	config := bigcache.DefaultConfig(10 * time.Minute)
```

Comment:
make them configurable so that we can tune the parameters later

Reply:
We will make the configuration later here